pkg/storer/reserve.go

```diff
@@ -172,69 +172,71 @@ func (db *DB) evictionWorker(ctx context.Context) {
 	}
 
 	cleanupExpired := func() {
-		db.metrics.ExpiryTriggersCount.Inc()
 		if !expirySem.TryAcquire(1) {
 			// if there is already a goroutine taking care of expirations dont wait
 			// for it to finish. Cleanup is called from all cases, so we dont
 			// need to trigger everytime.
 			return
 		}
-		defer expirySem.Release(1)
 
-		batchesToEvict, err := db.getExpiredBatches()
-		if err != nil {
-			db.logger.Error(err, "get expired batches")
-			return
-		}
-
-		if len(batchesToEvict) == 0 {
-			return
-		}
+		go func() {
+			defer expirySem.Release(1)
 
-		// we ensure unreserve is not running and if it is we cancel it and wait
-		// for it to finish, this is to prevent unreserve and expirations running
-		// at the same time. The expiration will free up space so the unreserve
-		// target might change. This is to prevent unreserve from running with
-		// the old target.
-		if !unreserveSem.TryAcquire(1) {
-			cancelUnreserve()
-			err := unreserveSem.Acquire(ctx, 1)
+			batchesToEvict, err := db.getExpiredBatches()
 			if err != nil {
-				db.logger.Error(err, "acquire unreserve semaphore")
+				db.logger.Error(err, "get expired batches")
 				return
 			}
-			// trigger it again at the end if required
-			defer db.events.Trigger(reserveOverCapacity)
-		}
-		defer unreserveSem.Release(1)
-
-		defer db.events.Trigger(reserveUnreserved)
 
-		db.metrics.ExpiryRunsCount.Inc()
-
-		for _, batchID := range batchesToEvict {
-			b := batchID
-			if err := expiryWorkers.Acquire(ctx, 1); err != nil {
-				db.logger.Error(err, "acquire expiry worker semaphore")
+			if len(batchesToEvict) == 0 {
 				return
 			}
-			go func() {
-				defer expiryWorkers.Release(1)
 
-				err := db.removeExpiredBatch(ctx, b)
+			// we ensure unreserve is not running and if it is we cancel it and wait
+			// for it to finish, this is to prevent unreserve and expirations running
+			// at the same time. The expiration will free up space so the unreserve
+			// target might change. This is to prevent unreserve from running with
+			// the old target.
+			if !unreserveSem.TryAcquire(1) {
+				cancelUnreserve()
+				err := unreserveSem.Acquire(ctx, 1)
 				if err != nil {
-					db.logger.Error(err, "remove expired batch", "batch_id", hex.EncodeToString(b))
-				} else {
-					db.metrics.ExpiredBatchCount.Inc()
+					db.logger.Error(err, "acquire unreserve semaphore")
+					return
 				}
-			}()
-		}
+				// trigger it again at the end if required
+				defer db.events.Trigger(reserveOverCapacity)
+			}
+			defer unreserveSem.Release(1)
 
-		if err := expiryWorkers.Acquire(ctx, 4); err != nil {
-			db.logger.Error(err, "wait for expiry workers")
-			return
-		}
-		expiryWorkers.Release(4)
+			defer db.events.Trigger(reserveUnreserved)
```

Comment:
why is this emitted?

Reply:
This was actually added for the test mainly. No one is listening for it. I thought its easier to use the same one than to have a separate key. Wdyt?

Reply:
right, works for me, maybe a small comment is needed.